lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -109,13 +117,40 @@ newTransactionLayer (Hash block0) = TransactionLayer
             $ Left ErrExceededInpsOrOuts
     }
 
+-- | Provide a transaction witness for a given private key. The type of witness
+-- if different between types of keys and, with backward-compatible support, we
+-- may have to support many types for one backend target.
```

Comment:
> The type of witness ~if~ different between types of keys

_The type of witness **is** different between *different* types of keys_

Question: would it be possible to use a the random address scheme with new funds, and put sequential addresses in legacy funds (ignoring the specifics of the current mainnet and coming testnet)? If it is possible, maybe the "why" of the cited statement could be made a bit clearer.

> we **may** have to support many types for one backend target 

Why "may"? We are?

Reply:
We are indeed. 

> would it be possible to use a the random address scheme with new funds
No, because the address structure for new funds makes it impossible to implement the random scheme. 

>  put sequential addresses in legacy funds
No. Jörmungandr doesn't allow that.

Reply:
extended comment to clarify that :+1: